core/src/main/java/org/mule/runtime/core/internal/client/DefaultLocalMuleClient.java

```diff
@@ -195,7 +196,10 @@ public void dispatch(String url, InternalMessage message, OperationOptions opera
   }
 
   protected Event createMuleEvent(InternalMessage message) throws MuleException {
-    return baseEventBuilder(message).build();
+    Event event = baseEventBuilder(message).build();
+    muleContext.getRegistry().lookupObject(StreamingManagerAdapter.class).registerEventContext(event.getContext());
```

Comment:
nope. And in case this turns up to be needed, the streaming manager should be injected on the client's initialisation. This lookup is expensive